src/main/java/org/semux/core/state/AccountStateImpl.java

```diff
@@ -104,28 +107,44 @@ public void adjustLocked(byte[] address, Amount delta) {
     }
 
     @Override
-    public void getCode(byte[] address) {
-        throw new UnsupportedOperationException("getCode() is not yet supported");
+    public byte[] getCode(byte[] address) {
+        Account acc = getAccount(address);
```

Comment:
We need specific rule on how and when accounts are being generated.

One rule can be: only create `Account` for mutation operations. Check the `Repository` documentation, or `RepositoryMockImp` for ref implementation.

Reply:
If we break code/storage into separate db tables as mentioned above, wouldn't this problem go away?